src/lib/i18n/index.es

```diff
@@ -0,0 +1,211 @@
+/**
+ * Default language.
+ * @type {string}
+ */
+const DEFAULT_LANGUAGE = 'en'
+
+/**
+ * Pattern to detect params in a string.
+ * @type {Regexp}
+ */
+let paramsPattern = /%(\d+)/g
+
+/**
+ * To store all the created instances.
+ * @type {Object}
+ */
+let instances = {}
+
+/**
+ * Current language. Shared by all the instances.
+ * @type {String}
+ */
+let language = DEFAULT_LANGUAGE
+
+/**
+ * Public API.
+ * @type {Object}
+ */
+class I18N {
+
+  /**
+   * Creates an i18n instance.
+   * If the ID is specified returns a singleton for each ID.
+   * If the ID is not specified returns a new instance.
+   * @param {string} id Instance ID. Optional.
+   */
+  constructor (id) {
+    let isValidId = typeof id === 'string'
+
+    if (!!isValidId && !!instances[id]) {
```

Comment:
With this code, you're making the engine to check twice the isValidId boolean.
It's easier by checking it only once.
```
if ( !!isValidId ) {
  if ( !!instances[id] ) {
    return instances[id];
  } else {
    instances[id] = this;
  } 
}
this._translations = {}
```

However, I wonder what happens when isValidIs is true but  instances[id] is already set. I am also wondering what's the correct behaviour when isValidId is false, since your code will return the I18N object with a _translations property inside.